src/java/org/apache/cassandra/cql3/QueryOptions.java

```diff
@@ -188,9 +190,10 @@ public PagingState getPagingState()
     }
 
     /**  Serial consistency for conditional updates. */
-    public ConsistencyLevel getSerialConsistency()
+    public ConsistencyLevel getSerialConsistency(@Nullable QueryState state)
```

Comment:
I think we should not pass `Nullable`  if we do not check for `null` for `state`

Reply:
Methods called from here also use `Nullable`  ... it's not until `Guardrail.enabled` is finally reached that `state` is checked for `null` value.
I'm not quite understanding what is suggested here - remove just this `Nullable` or this and all others leading to `Guardrail.enabled`? I always thought of `Nullable` as a sort of documentation that `null` is allowed here without needing to follow the call chain looking for the point it's actually checked.